Entities/Scene.cpp

```diff
@@ -916,9 +919,14 @@ int Scene::LoadData(bool placeObjects, bool initPathfinding, bool placeUnits)
     if (initPathfinding)
     {
         // Create the pathfinding stuff based on the current scene
-        m_pPathFinder = new PathFinder(this, 20, 2000);
-        // Update all the pathfinding data
-        m_pPathFinder->RecalculateAllCosts();
+		int sceneArea = GetWidth() * GetHeight();
+		int pathfinderGridNodeSize = g_SettingsMan.GetPathFinderGridNodeSize();
+		unsigned int numberOfBlocksToAllocate = 4000;//std::min(128000, sceneArea / (pathfinderGridNodeSize * pathfinderGridNodeSize));
+        
+        for (int i = 0; i < sc_pathfinderCount; ++i) {
```

Comment:
If you get rid of `sc_pathfinderCount` (see comment in the header file), you can use `m_pPathFinders.max_size()` instead. Iirc size won't work for ya, cause it cares about what's actually in the array, but max_size will.